pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex.go

```diff
@@ -848,17 +937,43 @@ func PushXds[T any](xds model.XDSUpdater, f func(T) model.ConfigKey) func(events
 	}
 }
 
-func PushXdsAddress[T any](xds model.XDSUpdater, f func(T) string) func(events []krt.Event[T]) {
+// Sometimes, the addresses we push depend on the event.
+func PushXdsAddressesWithEvent[T any](xds model.XDSUpdater, f func(krt.Event[T]) sets.String) func(events []krt.Event[T]) {
+	return func(events []krt.Event[T]) {
+		au := sets.New[string]()
+		for _, e := range events {
+			au = au.Union(f(e))
+		}
+		au = au.Delete("")
```

Comment:
Why?

Reply:
This is stuff from when I wasn't using krt of split horizon. WIll delete